symfony/plugins/orangehrmAuthenticationPlugin/Service/ResetPasswordService.php

```diff
@@ -0,0 +1,309 @@
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Authentication\Service;
+
+use OrangeHRM\Admin\Dto\UserSearchFilterParams;
+use OrangeHRM\Admin\Traits\Service\UserServiceTrait;
+use OrangeHRM\Authentication\Dao\ResetPasswordDao;
+use OrangeHRM\Authentication\Dto\UserCredential;
+use OrangeHRM\Config\Config;
+use OrangeHRM\Core\Service\EmailService;
+use OrangeHRM\Core\Traits\ControllerTrait;
+use OrangeHRM\Core\Traits\LoggerTrait;
+use OrangeHRM\Core\Traits\ORM\EntityManagerHelperTrait;
+use OrangeHRM\Core\Traits\Service\DateTimeHelperTrait;
+use OrangeHRM\Core\Utility\Base64Url;
+use OrangeHRM\Entity\EmailConfiguration;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Entity\ResetPassword;
+use OrangeHRM\Entity\User;
+use OrangeHRM\Framework\Routing\UrlGenerator;
+use OrangeHRM\Framework\Services;
+
+class ResetPasswordService
+{
+    use DateTimeHelperTrait;
+    use EntityManagerHelperTrait;
+    use LoggerTrait;
+    use ControllerTrait;
+    use UserServiceTrait;
+
+    public const RESET_PASSWORD_TOKEN_RANDOM_BYTES_LENGTH = 16;
+    protected ?ResetPasswordDao $resetPasswordDao = null;
+    protected ?EmailService $emailService = null;
+
+    /**
+     * @return EmailService
+     */
+    public function getEmailService(): EmailService
+    {
+        if (!$this->emailService instanceof EmailService) {
+            $this->emailService = new EmailService();
+        }
+        return $this->emailService;
+    }
+
+    /**
+     * @return ResetPasswordDao
+     */
+    public function getResetPasswordDao(): ResetPasswordDao
+    {
+        if (!$this->resetPasswordDao instanceof ResetPasswordDao) {
+            $this->resetPasswordDao = new ResetPasswordDao();
+        }
+        return $this->resetPasswordDao;
+    }
+
+    /**
+     * @param ResetPassword $resetPassword
+     * @return float
+     */
+    public function hasPasswordResetRequestNotExpired(ResetPassword $resetPassword): float
+    {
+        $strExpireTime = strtotime($resetPassword->getResetRequestDate()->format('Y-m-d H:i:s'));
+        $strCurrentTime = strtotime(date("Y-m-d H:i:s"));
+        return floor(($strCurrentTime - $strExpireTime) / (60 * 60 * 24));
```

Comment:
*

Reply:
check whether it's lesser than 1 then it's valid